### PR TITLE
Improvement/use pathlib

### PR DESCRIPTION
**Pull request by kvenkman**
Following #319, Replaced all things `os.path` to use `pathlib` instead 

----
📚 Documentation preview 📚: https://earthaccess--459.org.readthedocs.build/en/459/

### CODE REVIEW

File: earthaccess/store.py [hunk 1]
```diff
@@ -526,7 +526,7 @@ def _get_urls(
             # TODO: make this parallel or concurrent
             for file in data_links:
                 s3_fs.get(file, local_path)
-                file_name = os.path.join(local_path, os.path.basename(file))
+                file_name = Path(local_path, Path(file).name)
```
Comment (jhkennedy) [inline earthaccess/store.py:529]:
```suggestion
                file_name = Path(local_path) / Path(file).name
```

File: earthaccess/store.py [hunk 2]
```diff
@@ -572,7 +572,7 @@ def _get_granules(
             # TODO: make this async
             for file in data_links:
                 s3_fs.get(file, local_path)
-                file_name = os.path.join(local_path, os.path.basename(file))
+                file_name = Path(local_path, Path(file).name)
```
Comment (jhkennedy) [inline earthaccess/store.py:575]:
```suggestion
                file_name = Path(local_path) / Path(file).name
```

File: earthaccess/store.py [hunk 3]
```diff
@@ -466,7 +466,7 @@ def get(
             List of downloaded files
         """
         if local_path is None:
-            local_path = os.path.join(
+            local_path = Path(
```
Comment (jhkennedy) [inline earthaccess/store.py:469]: `local_path`, when passed to any of the `_get` dispatch methods, is expected to be a string.

Either we should change the `_get` methods to take a Path or this needs to be made a string.

Comment (jhkennedy) [reply earthaccess/store.py:469]: Switching the called methods to taking a Path object looks straightforward, so I'd lean that way.

**Review by jhkennedy — CHANGES_REQUESTED** [review]
@kvenkman looks good! I've added some suggestions and there's one change we'll want to make -- do you want to make the changes or want me to push changes?

File: earthaccess/store.py (hunk 3 above)
Comment (mfisher87) [reply earthaccess/store.py:469]: >Switching the called methods to taking a Path object looks straightforward, so I'd lean that way.

I like this too :)

Comment (kvenkman) [reply earthaccess/store.py:469]: I'll update the methods to accept a Path objects - thanks for pointing that out!

**Comment by kvenkman** [comment]
Thanks for the suggestions @jhkennedy - I've folded them in :smile:

File: earthaccess/store.py (hunk 3 above)
Comment (mfisher87) [reply earthaccess/store.py:469]: Thanks for your work making _earthaccess_ better :)

**Comment by mfisher87** [comment]
pre-commit.ci autofix

**Review by jhkennedy — APPROVED** [review]
I think this is g2g now! Thanks @kvenkman . 

I pushed a couple of minor changes to get the tests to all pass:
https://github.com/nsidc/earthaccess/compare/c68e634..453e7f9

**Pull request merged**